src/messaging.cc

```diff
@@ -483,7 +483,10 @@ bool ExternalSystemMessageHandler::send(int pid, int type, void* data, int lengt
   }
 
   uint8* buffer = unvoid_cast<uint8*>(malloc(buffer_size));
-  if (buffer == null) return false;
+  if (buffer == null) {
+    free(data);
```

Comment:
Not sure that freeing the data on allocation error is the best approach. As a consumer of the Api, if this returns false, the most obvious remediation action would be to retry the operation after doing other corrective actions (like invoking the GC). If the data is free'ed then the client code needs to reallocate the memory and copy over the payload.

Reply:
I agree. Maybe it would be better to have a flag that controls this (a boolean passed to send, maybe with a reasonable default to not deallocate on errors)?